js/modules/Forms/Condition/Engine.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2025 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/* global glpi_toast_error, getAjaxCsrfToken */
+
+export class GlpiFormConditionEngine
+{
+    #form_id;
+
+    constructor(form_id)
+    {
+        this.#form_id = form_id;
+    }
+
+    async computeVisiblity(container)
+    {
+        container = document.querySelector(container);
+
+        try {
+            // Send data to server for computation and apply results.
+            const results = await this.#computeVisibilityOnBackend({
+                answers: this.#getQuestionsData(container)
+            });
+            this.#applyVisibilityResults(results, container);
+        } catch (e) {
+            console.error(e);
+            glpi_toast_error(
+                __("Unexpected error, please contact your administrator.")
```

Comment:
Not related to this PR, but I think the `please contact your administrator` text is too directive. I guess a `An unexpected error occurred.` message would be enough.